app/controllers/application_controller.rb

```diff
@@ -2,4 +2,24 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  def widget_params
+    # read all json schemas
+    jsons = Dir[Rails.root.join('db','json','*.json_schema')].map{ |f| JSON.parse File.read(f) }
+    allowed_keys = []
+
+    # add allowed keys as fields, dictionaries, or hashes depending on their type
+    jsons.map{|j| j['properties'] }.each do |properties|
```

Comment:
Will this handle nested properties? Like, for instance, if I had a property I wanted to create which looked like this:

```
foo: {
  bar: [],
  bat: 1,
  baz: {
    thing: 'string'
  }
}```

Would this loop be able to handle that?

To be clear, I'm not sure that we actually want to do that, though if it doesn't handle that I think it probably makes sense to add a quick note that we're intentionally not handling nested types so that if we ever do want to do that, whoever tries to add it and can't figure out why it's not working will have a brief explanation to help them remember.